cellmlmanip/model.py

```diff
@@ -0,0 +1,574 @@
+                                variable.assigned_to,
+                                source_dummy.assigned_to)
+        return is_okay
+
+    def check_variables_in_equations(self):
+        """Every variable we have should have been used in an equation."""
+        pass
+
+    def connect_variables(self, source_variable: str, target_variable: str):
+        """Given the source and target component and variable, create a connection by assigning
+        the symbol from the source to the target. If units are not the same, it will add an equation
+        to the target component reflecting the relationship. If a symbol has not been assigned to
+        the source variable, then return False.
+
+        :param source_variable: source variable name
+        :param target_variable: target variable name
+        """
+        logger.debug('connect_variables(%s ⟶ %s)', source_variable, target_variable)
+
+        source_variable = self.get_meta_dummy(source_variable)
+        target_variable = self.get_meta_dummy(target_variable)
+
+        # If the source variable has already been assigned a final symbol
+        if source_variable.assigned_to:
+
+            if target_variable.assigned_to:
+                raise ValueError('Target already assigned to %s before assignment to %s' %
+                                 (target_variable.assigned_to, source_variable.assigned_to))
+
+            # If source/target variable is in the same unit
+            if source_variable.units == target_variable.units:
+                # Direct substitution is possible
+                target_variable.assigned_to = source_variable.assigned_to
+                # everywhere the target variable is used, replace with source variable
+                for index, equation in enumerate(self.equations):
+                    self.equations[index] = equation.xreplace(
+                        {target_variable.dummy: source_variable.assigned_to}
+                    )
+            else:
+                # Requires a conversion, so we add an equation assigning the target dummy variable
+                # to the source variable
+                self.equations.append(
+                    # TODO: do unit conversion here?
+                    sympy.Eq(target_variable.dummy, source_variable.assigned_to)
+                )
+                logger.info('Connection req. unit conversion: %s', self.equations[-1])
+
+                # The assigned symbol for this variable is itself
+                target_variable.assigned_to = target_variable.dummy
+            logger.debug('Updated target: %s', target_variable)
+            return True
+        # The source variable has not been assigned a symbol, so we can't make this connection
+        logger.info('The source variable has not been assigned to a symbol '
+                    '(i.e. expecting a connection): %s ⟶ %s',
+                    target_variable.name, source_variable.name)
+        return False
+
+    def add_rdf(self, rdf: str):
```

Comment:
Should these be alphabetic? Or have we not decided on that @jonc125 ?

Reply:
What do you mean?

Reply:
A lot of people do the methods in a class alphabetically.

Reply:
Oh, I see. I'm not strict on that, provided the order makes some sort of sense. I'll tend to search when I'm looking for something anyway! So having related methods adjacent (where you're likely to need to look at both when reading/changing the code) tends to be what I go for.

Reply:
Cool! That's what I used to do but it turns out to be difficult to agree upon between people, and moving stuff around messes up your git history :D
(By which I mean, if you change _and move_ a method in a single commit, it looks like delete/insert, so git can't point to the modified lines anymore)

Reply:
Yeah, don't do that :)

First/lead author of the code gets to decide. Anyone else doesn't get to move methods except with agreement, and in a separate commit.